### PR TITLE
node: improve pprof lifecycle

### PR DESCRIPTION
**Pull request by cmwaters**
This PR does a few minor changes.
- It replicates the logic of the prometheus server but applies it to pprof so that `OnStop` actually shuts down the node
- It removes the unusued `id.go` file
- It creates a new `setup.go` file and moves the setup logic there to make `node.go` more easier to understand

Addresses: https://github.com/tendermint/tendermint/issues/9627

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -1235,6 +670,22 @@ func (n *Node) startPrometheusServer(addr string) *http.Server {
 	return srv
 }
 
+// starts a ppro
+func (n *Node) startPprofServer() *http.Server {
+	srv := &http.Server{
+		Addr:              n.config.RPC.PprofListenAddress,
+		Handler:           nil,
```
Comment (williambanfield) [inline node/node.go:677]: Why is the handler `nil`?

File: node/node.go [hunk 2]
```diff
@@ -1235,6 +670,22 @@ func (n *Node) startPrometheusServer(addr string) *http.Server {
...
+		ReadHeaderTimeout: readHeaderTimeout,
+	}
+	go func() {
+		if err := srv.ListenAndServe(); err != http.ErrServerClosed {
+			// Error starting or closing listener:
+			n.Logger.Error("Prometheus HTTP server ListenAndServe", "err", err)
```
Comment (williambanfield) [inline node/node.go:683]:
```suggestion
			n.Logger.Error("pprof HTTP server ListenAndServe", "err", err)
```

File: node/node.go (hunk 1 above)
Comment (williambanfield) [reply node/node.go:677]: Interesting. It looks like the `pprof` library import was removed here, so I'm not sure `pprof` is being linked in at all in this code.

Comment (cmwaters) [reply node/node.go:677]: Ah lol I removed that because `gosec` said: "Profiling endpoint is automatically exposed on /debug/pprof"

Comment (cmwaters) [reply node/node.go:677]: I added a test (`TestPprofServer`) that seems to be passing

File: node/node_test.go [hunk 3]
```diff
@@ -135,6 +136,25 @@ func TestNodeSetAppVersion(t *testing.T) {
...
+	_, err := http.Get("http://" + config.RPC.PprofListenAddress)
+	assert.Error(t, err)
+
+	n, err := DefaultNewNode(config, log.TestingLogger())
+	assert.NoError(t, err)
+	assert.NoError(t, n.Start())
+	defer n.Stop()
+	assert.NotNil(t, n.pprofSrv)
+
+	_, err = http.Get("http://" + config.RPC.PprofListenAddress)
```
Comment (williambanfield) [inline node/node_test.go:154]: This just tests that there is _a_ server running, not that it's serving the pprof data. I'd add a check that the response doesn't return `404` on `/debug/pprof/`.

Comment (cmwaters) [reply node/node_test.go:154]: good point

File: node/node_test.go [hunk 4]
```diff
@@ -135,6 +136,29 @@ func TestNodeSetAppVersion(t *testing.T) {
...
+	assert.NoError(t, n.Start())
+	defer func() {
+		require.NoError(t, n.Stop())
+	}()
+	assert.NotNil(t, n.pprofSrv)
+
+	resp, err := http.Get("http://" + config.RPC.PprofListenAddress + "/debug/pprof")
+	assert.NoError(t, err)
+	defer resp.Body.Close()
+	assert.Equal(t, "200 OK", resp.Status)
```
Comment (williambanfield) [inline node/node_test.go:159]: I'd probably use `resp.StatusCode`

**Review by williambanfield — APPROVED** [review]

File: node/node.go [hunk 5]
```diff
@@ -1215,9 +652,9 @@ func (n *Node) startRPC() ([]net.Listener, error) {
 
 // startPrometheusServer starts a Prometheus HTTP server, listening for metrics
 // collectors on addr.
```
Comment (thanethomson) [inline node/node.go:654]: The `addr` field was removed, so this comment should be updated.

File: node/node.go [hunk 6]
```diff
@@ -1235,6 +672,22 @@ func (n *Node) startPrometheusServer(addr string) *http.Server {
 	return srv
 }
 
+// starts a ppro
```
Comment (thanethomson) [inline node/node.go:675]: Incomplete comment?

**Review by thanethomson — COMMENTED** [review]
While I like the fact that this PR seems to be making the node package a little more readable, it seems as though there are two substantial changes lumped in together in this PR under a title that only points to one of them:
1. Makes changes to the pprof lifecycle. How precisely are these an improvement? It's not obvious from the description in the PR.
2. Refactors the node package to be more readable.

I wouldn't have approved this PR and would've rather advocated for separating out these changes as two separate PRs. Having them lumped together makes tracing changes far harder when one's doing code archeology.

**Comment by cmwaters** [comment]
> While I like the fact that this PR seems to be making the node package a little more readable, it seems as though there are two substantial changes lumped in together in this PR under a title that only points to one of them:

Hmm I don't see either of these as substantial changes. In fact I'm working on a larger PR on node start up phases and actually peeled these changes off that work. Regardless of the magnitude of this work, you're correct in that there is 3 separate units of logical work done in this PR but in terms of code archeology I have outlined these in the PR description. If you're unhappy with the title, I am happy to change it

**Comment by thanethomson** [comment]
> If you're unhappy with the title, I am happy to change it

Separating out distinct pieces of work into separate PRs is much more important than just the title of this PR.

**Pull request merged**